File: backend/cal.py
from flask import Flask, request, jsonify
from flask_cors import CORS

app = Flask(__name__)
CORS(app)

@app.route("/")
def home():
    return "Calculator Backend is Running"

@app.route("/calculate", methods=["POST"])
def calculate():
    data = request.get_json()

    num1 = float(data["num1"])
    num2 = float(data["num2"])
    operation = data["operation"]

    if operation == "add":
        result = num1 + num2
    elif operation == "subtract":
        result = num1 - num2
    elif operation == "multiply":
        result = num1 * num2
    elif operation == "divide":
        if num2 == 0:
            return jsonify({"answer": "Cannot divide by zero"})
        result = num1 / num2
    else:
        return jsonify({"answer": "Invalid operation"})

    return jsonify({"answer": result})

if __name__ == "__main__":
    app.run(host="0.0.0.0", port=5000)

File: cal.js
function calculate() {

    let num1 = parseFloat(document.getElementById("num1").value);
    let num2 = parseFloat(document.getElementById("num2").value);
    let operator = document.getElementById("operator").value;

    if (isNaN(num1) || isNaN(num2)) {
        document.getElementById("result").innerHTML = "Answer: Enter both numbers";
        return;
    }

    let answer;

    if (operator == "+") {
        answer = num1 + num2;
    }
    else if (operator == "-") {
        answer = num1 - num2;
    }
    else if (operator == "*") {
        answer = num1 * num2;
    }
    else if (operator == "/") {

        if (num2 == 0) {
            answer = "Cannot divide by zero";
        } else {
            answer = num1 / num2;
        }
    }

    document.getElementById("result").innerHTML = "Answer: " + answer;
}
